### PR TITLE
Did it properly this time, added csharp guide

### PR DESCRIPTION
**Pull request by SylveonDeko**
## What does this Pull Request change?
Added a csharp guide
## I agree that my Pull Request:

- Follows the [Contribution Guidelines](https://github.com/nekos-best/docs#contribution-guidelines)
- Follows [GitHub's Community Guidelines](https://docs.github.com/en/github/site-policy/github-community-guidelines)
- Contains an example that is well documented and is ordered alphabetically in [SUMMARY.md](https://github.com/nekos-best/docs/blob/main/src/SUMMARY.md)
- Contains an example that is not a duplicate of an already existing example.

## Any additional notes?

None

### CODE REVIEW

**Comment by oriel-beck** [comment]
Please update your example and your package's readme to reflect 1.0.2's namespace change

**Comment by SylveonDeko** [comment]
done lol

File: src/examples/unofficial/csharp.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+public class ExampleClass
+{
+  private readonly NekosBestApi _nekosBestApi;
+
+  public ExampleClass() 
+    => _nekosBestApi = new NekosBestApi();
+
+  public async Task Hug() 
+  {
+    var image = await _nekosBestApi.ActionApi.Hug();
```
Comment (thund-x) [inline src/examples/unofficial/csharp.md]:
```suggestion
    var image = await _nekosBestApi.ActionsApi.Hug();
```

You are missing an `s` here

File: src/examples/unofficial/csharp.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+You can provide your own HttpClient instance, but you have to set the BaseAddress manually
+
+```cs
+using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using NekosBestApiNet;
+
+public ExampleClass()
```
Comment (thund-x) [inline src/examples/unofficial/csharp.md]: You forgot to wrap the constructor to the class `ExampleClass`

**Review by thund-x — CHANGES_REQUESTED** [review]

File: src/examples/unofficial/csharp.md (hunk 2 above)
Comment (SylveonDeko) [reply src/examples/unofficial/csharp.md]: Didn't mean to put () there

File: src/examples/unofficial/csharp.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+public ExampleClass
+{
+  var httpClient = new HttpClient 
+  {
+    BaseAddress = new Uri("https://nekos.best/api/v2")
+  };
+
+  _nekosBestApi = new NekosBestApi(httpClient);
+}
```
Comment (thund-x) [inline src/examples/unofficial/csharp.md]:
```suggestion
class ExampleClass
{
    private NekosBestApi _nekosBestApi;

    public ExampleClass()
    {
        var httpClient = new HttpClient
        {
            BaseAddress = new Uri("https://nekos.best/api/v2")
        };

        _nekosBestApi = new NekosBestApi(httpClient);
    }
}```

File: src/examples/unofficial/csharp.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+    ConfigureServices(services);
+    _serviceProvider = services.BuildServiceProvider();
+  }
+
+  public async Task RunAsync() 
+  {
+    var exampleClass = _serviceProvider.GetService<ExampleClass>();
+
+    var image = await _nekosBestApi.ActionApi.Hug();
```
Comment (thund-x) [inline src/examples/unofficial/csharp.md]:
```suggestion
    var image = await _nekosBestApi.ActionsApi.Hug();
```

File: src/examples/unofficial/csharp.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+### Example using dependency injection
+
+Create a ServiceCollection, then add an instance of the NekosBestApi class to it
+
+
+```cs
+using System;
+using System.Threading.Tasks;
+using NekosBestApiNet;
+using Microsoft.Extensions.DependencyInjection;
```
Comment (thund-x) [inline src/examples/unofficial/csharp.md:75]: Perhaps you should mention that this assembly reference is provided by the package `Microsoft.Extensions.DependencyInjection`

**Review by thund-x — CHANGES_REQUESTED** [review]

File: src/examples/unofficial/csharp.md (hunk 5 above)
Comment (SylveonDeko) [reply src/examples/unofficial/csharp.md:75]: I'm pretty sure they would already know this if using dependency injection

**Review by thund-x — APPROVED** [review]

**Comment by Thusuzzee** [comment]
Thanks for your contribution.

**Pull request merged**